pkg/reconciler/topology/partitionset/resources.go

```diff
@@ -35,6 +36,7 @@ func generatePartition(name string, matchExpressions []metav1.LabelSelectorRequi
 	for _, label := range labels {
 		pname = pname + "-" + strings.ToLower(matchLabels[label])
 	}
+	pname = pname[:min(validation.DNS1123SubdomainMaxLength-1, len(pname))]
```

Comment:
so, there is a limit :)

Reply:
please add a unit test

Reply:
What about collisions? This is why we use the hash in many other places.

Reply:
GenerateName is used to avoid collisions.